megamek/src/megamek/client/bot/princess/SwarmContext.java

```diff
@@ -0,0 +1,360 @@
+/*
+ * Copyright (c) 2025 - The MegaMek Team. All Rights Reserved.
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by the Free
+ * Software Foundation; either version 2 of the License, or (at your option)
+ * any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+ * or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License
+ * for more details.
+ *
+ */
+package megamek.client.bot.princess;
+
+import megamek.common.*;
+import megamek.common.annotations.Nullable;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+/**
+ * Context class to control the units as a swarm.
+ * Contains information about the current state of the game
+ */
+public class SwarmContext {
+
+    private final Map<Integer, Integer> enemyTargetCounts = new HashMap<>();
+    private final Map<Coords, Integer> positionDensity = new HashMap<>();
+    private final List<Coords> strategicGoals = new Vector<>();
+    private Coords currentCenter;
+
+    private int quadrantHeight = 0;
+    private int quadrantWidth = 0;
+    private int offsetX = 0;
+    private int offsetY = 0;
+
+    private int clusterUnitsSize = 0;
+
+    private final Map<Integer, SwarmCluster> unitClusters = new HashMap<>();
+    private final List<SwarmCluster> clusters = new ArrayList<>();
+
+    public SwarmContext() {;
+    }
+
+    /**
+     * Record an enemy target
+     * @param enemyId The enemy id
+     */
+    public void recordEnemyTarget(int enemyId) {
+        enemyTargetCounts.put(enemyId, enemyTargetCounts.getOrDefault(enemyId, 0) + 1);
```

Comment:
Why the +1?

Reply:
its increasing the number of "units targeting this enemy", thats what it means by "record enemy target", I am improving the javadoc on this